src/flexspi.rs

```diff
@@ -0,0 +1,298 @@
+//!     .instr(Instr::new(CMD, Pads::One, READ_STATUS_REGISTER_1))
+//!     .instr(Instr::new(READ, Pads::One, 0x04))
+//!     .build();
+//! ```
+//!
+//! Then, assign each sequence to a [`Command`] in a `LookupTable`:
+//!
+//! ```
+//! use imxrt_boot_gen::flexspi::{Command, LookupTable};
+//! # use imxrt_boot_gen::flexspi::{Sequence, SequenceBuilder};
+//!
+//! # const SEQ_READ: Sequence = SequenceBuilder::new().build();
+//! # const SEQ_READ_STATUS: Sequence = SequenceBuilder::new().build();
+//! const LUT: LookupTable = LookupTable::new()
+//!     .command(Command::Read, SEQ_READ)
+//!     .command(Command::ReadStatus, SEQ_READ_STATUS);
+//! ```
+//!
+//! # FlexSPI Configuration Block
+//!
+//! Once you've created your sequences and lookup table, use the lookup table to create
+//! a [`ConfigurationBlock`]. See the `ConfigurationBlock` documentation
+//! for more information.
+
+mod fields;
+mod lookup;
+mod sequence;
+
+pub use fields::*;
+pub use lookup::{Command, LookupTable};
+pub use sequence::{opcodes, Instr, Pads, Sequence, SequenceBuilder, JUMP_ON_CS, STOP};
+
+/// ASCII 'FCFB'
+const TAG: u32 = 0x4246_4346;
+/// [07:00] bugfix = 0
+/// [15:08] minor
+/// [23:16] major = 1
+/// [31:24] ascii ‘V’
+const VERSION: u32 = 0x5601_0000;
+
+/// The recommended `csHoldTime`, `0x03`.
+///
+/// This is the default value if not set with [`ConfigurationBlock::cs_hold_time`].
+pub const RECOMMENDED_CS_HOLD_TIME: u8 = 0x03;
+/// The recommended `csSetupTime`, `0x03`.
+///
+/// This is the default value if not set with [`ConfigurationBlock::cs_setup_time`].
+pub const RECOMMENDED_CS_SETUP_TIME: u8 = 0x03;
+
+/// FlexSPI configuration block
+///
+/// The FlexSPI configuration block consists of parameters that are for specific flash
+/// devices. The configuration block includes the FlexSPI [`LookupTable`]. The configuration
+/// block is shared between serial NOR and NAND configuration blocks.
+///
+/// # Default Values
+///
+/// - `cs_hold_time` is [`RECOMMENDED_CS_HOLD_TIME`]
+/// - `cs_setup_time` is [`RECOMMENDED_CS_SETUP_TIME`]
+///
+/// All other configurable values are set to a bit pattern of 0.
+///
+/// # Examples
+///
+/// ```
+/// use imxrt_boot_gen::flexspi::*;
+///
+/// # const LUT: LookupTable = LookupTable::new();
+/// const FLEXSPI_CONFIGURATION_BLOCK: ConfigurationBlock =
+///     ConfigurationBlock::new(LUT)
+///         .read_sample_clk_src(ReadSampleClockSource::LoopbackFromDQSPad)
+///         .cs_hold_time(0x01)
+///         .cs_setup_time(0x02)
+///         .column_address_width(ColumnAddressWidth::OtherDevices)
+///         .device_mode_configuration(DeviceModeConfiguration::Disabled)
+///         .wait_time_cfg_commands(WaitTimeConfigurationCommands::new(40_000))
+///         .flash_size(SerialFlashRegion::A1, 0x0020_0000)
+///         .serial_clk_freq(SerialClockFrequency::MHz60)
+///         .serial_flash_pad_type(FlashPadType::Quad);
+///
+#[derive(Debug, Clone, Copy)]
+#[repr(C, packed)]
```

Comment:
It's nice that you are able to to use repr(C, packed) here to effectively name the various memory sections in the configuration block here